### PR TITLE
Feature/bump hikari

### PR DESCRIPTION
**Pull request by jboreiko**
@amlingpalantir

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/662)

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -26,6 +26,22 @@ Changelog
...
+
+.. list-table::
+    :widths: 5 40
+    :header-rows: 1
+
+    *    - Type
+         - Change
+
+    *    - |changed|
+         - Updated HikariCP dependency from 2.4.3 to 2.4.7 to complie with updates in internal products
```
Comment (schlosna) [inline docs/source/release_notes/release-notes.rst]: spelling nit: comply

**Comment by schlosna** [comment]
Probably worth noting the changes since 2.4.3 via https://github.com/brettwooldridge/HikariCP/blob/dev/CHANGES

```
HikariCP Changes

Changes in 2.4.7

 * Miscellaneous stability improvements.

 * Removed Oracle SQL state 61000, added specific error code (2399) to evict connections
   when it is encountered.

 * issue 664: do not recycle PoolEntry objects that have closed their held connection.

 * issue 641, 643: reflection used method String.toUpperCase() without a Locale, which causes
   problems in some locales such as Turkish.

 * pull 632: added support for Prometheus metrics tracker.

 * issue 650: detect Amazon Redshift connection refused error codes.

Changes in 2.4.6

 * Added Oracle SQL error code 61000 (exceeded maximum connect time) to evict connections
   when it is encountered.

 * issue 621: fix NPE in shutdown if invoked during initialization.

 * issue 606, 610: housekeeper thread was running before all class members were
   initialized, causing an unrecoverable exception and disabling idle connection
   retirement (maximumLifetime still applied).

Changes in 2.4.5

 * issue 596: fix bug that occurs when minimumIdle is set to 0, but maximumPoolSize is
   not explicitly set.

 * issue 594: fix incompatibility with various libraries caused by storing a non-String
   object in System properties.

 * issue 593: improve logging when network timeout is not supported by the driver

 * issue 591: improve thread-safety of Statement proxies

Changes in 2.4.4

 * Generate unique sequential pool names, even across container classloaders to avoid
   JMX name collisions when registering pools.

 * Improve pool stability when running on computers using power-saving or sleep modes
   where wake-up previously caused pool to grow to maximum size.

 * Improve pool stability under severe thread-starvation conditions.  Previous code
   could interpret prolonged starvation (exceeding one minute) as non-contiguous clock
   advancement, and reacted by soft-evicting connections (unnecessarily).

 * Added connection timeout rate to Dropwizard metrics (ConnectionTimeoutRate).

 * issue 563: Do not start the house-keeping thread until after pool initialisation
   has succeeded.

 * issue 559: Ensure the pool refill after house-keeping does not enqueue more add
   connection requests if there are already minimumIdle requests pending.

 * issue 555: include support for Java 8 interface 'default' methods during proxy
   generation.

 * issue 547: decreased allowable minimum connectionTimeout and validationTimeout to
   250ms.

 * issue 495: implemented iterator() method on custom FastList to support Tomcat
   memory leak detection.

```

**Comment by schlosna** [comment]
Reviewed 2 of 2 files at r1, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/662)*

**Comment by jboreiko** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[docs/source/release_notes/release-notes.rst, line 40 \[r2\]](https://reviewable.io:443/reviews/palantir/atlasdb/662#-KMPpgdfLRQG6rqOKuHf-r2-40:-KMQBtDQyx3xHLf5EfkJ:-499146009) ([raw file](https://github.com/palantir/atlasdb/blob/6108989ceed4b5549257c11b8e3459f58af58242/docs/source/release_notes/release-notes.rst#L40)):*
<details><summary><i>Previously, schlosna (David Schlosnagle) wrote…</i></summary>
> spelling nit: comply
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/662)*

**Pull request merged**